src/functions/findFirst.js

```diff
@@ -48,12 +48,18 @@ export const findFirst = wrap({
         let search = asSequence(sequences[1]);
         const compareFunc = compare || constants.defaults.comparisonFunction;
 
-        // Handle empty or unbounded search subject
-        if(search.done() || search.unbounded()){
+        // simply return undefined if the search input is empty
+        if (search.done()){
+            return undefined;
+        }
+
+        // ... however throw a NotBoundedError if the search input is unbounded
+        if(search.unbounded()){
             throw NotBoundedError(search, {
                 message: "Search sequence is either empty or unbounded"
```

Comment:
Update the error message - ought to read "Failed to find substring", the error constructor adds to the message an explanation as to why. (the sequence was unbounded, was not known to be bounded, whatever)